### PR TITLE
New: Add support for enabling and disabling zoom (fixes #62)

### LINKED ISSUE #62: Option to disable interaction
Is there any way in which I can disable interaction with the graph? No zoom/pan and no drag of nodes 

Thanks

### PR DESCRIPTION
**Pull request by parmar-abhinav**
**Description:**

This pull request introduces enhancements related to zooming functionality. The changes aim to provide users with more control over zooming interactions. Here's an overview of the modifications made:

- Added a new property `isZoomEnabled` to the optional interaction object.

- `isZoomEnabled` allows users to enable or disable the zooming feature in the graph.

- When `isZoomEnabled` is set to `true`, users can interactively zoom in and out of the graph.

- When `isZoomEnabled` is set to `false`, zooming functionality is disabled, restricting users from changing the zoom level.

**Example:**

```typescript
//To enable zooming
orb.setSettings({ interaction: { isZoomEnabled: true } });
```

```typescript
//To disable zooming
orb.setSettings({ interaction: { isZoomEnabled: false } });
```

### CODE REVIEW

File: docs/view-default.md [hunk 1]
```diff
@@ -321,15 +321,17 @@ orb.events.on(OrbEventType.MOUSE_CLICK, (event) => {
 
 ### Property `interaction`
 
-The optional property `interaction` has one property that you can enable/disable:
+The optional property `interaction` has two properties that you can enable/disable:
```
Comment (tonilastre) [inline docs/view-default.md:324]: In this file, lines `:157` and `:85` are missing the mention of the `isZoomEnabled`.

File: src/views/orb-view.ts [hunk 2]
```diff
@@ -346,6 +355,10 @@ export class OrbView<N extends INodeBase, E extends IEdgeBase> implements IOrbVi
   };
 
   zoomed = (event: D3ZoomEvent<any, any>) => {
+    // If zoom is disabled then return
```
Comment (tonilastre) [inline src/views/orb-view.ts:358]: Btw it wasn't required for the drag because nodes can't be dragged, but the zoom interaction disable/enable should be implemented on the `orb-view-map`. This can be another PR.

**Review by tonilastre — CHANGES_REQUESTED** [review]
A small note in the docs missing. Everything else is good!

File: docs/view-default.md (hunk 1 above)
Comment (parmar-abhinav) [reply docs/view-default.md:326]: Added `isZoomEnabled` pls check

File: src/views/orb-view.ts (hunk 2 above)
Comment (parmar-abhinav) [reply src/views/orb-view.ts:358]: Sure will raise a new PR related to changes in `orb-view-map`.

File: src/views/orb-view.ts [hunk 3]
```diff
@@ -230,6 +232,13 @@ export class OrbView<N extends INodeBase, E extends IEdgeBase> implements IOrbVi
         this._settings.interaction.isDragEnabled =
           settings.interaction.isDragEnabled;
       }
+
+      // Check if isZoomEnabled is a boolean value
```
Comment (tonilastre) [inline src/views/orb-view.ts:236]: Btw you can remove these comments (including the ones for the drag check). It is usually good practice to just have a comment when you need to describe a complex part of the code. The code here is already well written so it doesn't need comments.

**Review by tonilastre — CHANGES_REQUESTED** [review]
There is one additional bug with the recenter + zoom. Check the suggestion and other comments regarding inline comments.

Btw, I am not sure why `husky` wasn't working on your machine, but please run `npm run lint:fix` because there are some lint issues in the code :) This will automatically fix them so you can commit them afterward.
